### PR TITLE
Add mutation testing workflow

### PR DESCRIPTION
**Pull request by sanmai**
### What and why? 🤔

Mutation testing is a way to detect faulty tests by adding slight but substantial changes ("mutants") to the source code until tests notice the changes by failing. If a test covering a line didn't notice the change, we call this mutant escaped.  In other words, the more escaped mutants you have, the less comprehensive is the test suite.

Although there are many ways to introduce mutants ("mutators"), we are starting with the most basic and obvious: [method and function call removal mutators](https://infection.github.io/guide/profiles.html#removal).

### Testing Steps ✍️

```
make mt SILENT=
```
Should show you the list of escaped mutants (changes the test didn't detect). You should also see this in the changed files section on this and every PR hereafter.

### You're it! 👑

Tag a random reviewer by adding `@Automattic/stream-builders` to the reviewers section, but mention them here for visibility as well.

### CODE REVIEW

File: .github/workflows/mt.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# yamllint disable rule:line-length
+# yamllint disable rule:braces
+
+name: Mutation Testing
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
```
Comment (sanmai) [inline .github/workflows/mt.yml]:
```suggestion
```

File: .github/workflows/mt.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# yamllint disable rule:line-length
+# yamllint disable rule:braces
+
+name: Mutation Testing
+
+on:
+  pull_request:
+
+permissions: read-all
+
```
Comment (sanmai) [inline .github/workflows/mt.yml]:
```suggestion

```

**Review by marco-altran — APPROVED** [review]
LGTM!

I am only worried if this would not cause pollution by showing warnings that are not related to the PR. Is this a concern?

**Comment by sanmai** [comment]
They will be shown, but under the diff of a PR. Line on this one. So no harm is expected.

**Pull request merged**